hr_attendance_ip_check/models/hr_employee.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from odoo import models, _
+from odoo.http import request
+from werkzeug.exceptions import HTTPException
+
+_logger = logging.getLogger(__name__)
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    def _get_current_ip(self):
+        """Get the current IP address from the request."""
+        try:
+            if request and request.httprequest:
+                return request.httprequest.remote_addr
+        except (AttributeError, HTTPException) as e:
+            _logger.error("Error getting IP address: %s", str(e), exc_info=True)
+        return None
+
+    def _validate_ip_address(self, action_type='attendance'):
+        """Validate if current IP is allowed for attendance actions."""
+        if not self.env['ir.config_parameter'].sudo().get_param(
```

Comment:
Make use of const_eval(after importing it)
from odoo.tools.safe_eval import const_eval

const_eval(self.env['ir.config_parameter'].sudo().get_param('hr_attendance.ip_check_enabled', 'False')))